backend_tv/lifter.cpp

```diff
@@ -6785,6 +6804,77 @@ class arm2llvm {
       updateOutputReg(res);
       break;
     }
+    case AArch64::SHADDv8i8:
+    case AArch64::SHADDv16i8:
+    case AArch64::SHADDv4i16:
+    case AArch64::SHADDv8i16:
+    case AArch64::SHADDv2i32:
+    case AArch64::SHADDv4i32:
+      //    case AArch64::SHSUBv8i8:
+      //    case AArch64::SHSUBv16i8:
+      //    case AArch64::SHSUBv4i16:
+      //    case AArch64::SHSUBv8i16:
+      //    case AArch64::SHSUBv2i32:
+      //    case AArch64::SHSUBv4i32:
+      {
+        int numElts, eltSize;
```

Comment:
please use one of the macros instead of explicitly writing this big switch each time! GET_SIZES6 should work fine for you, just search for it in lifter.cpp

Reply:
looks great except for the one thing

Reply:
Oh I saw that and planned on using it. But I did not use it because the operation of SHSUB and SHADD is the same. Merging those meant I could not use the macro. 

I commented the SUB part because the tests I found had Neon assembly intrinsic. Maybe the fuzzer can find tests so left it uncommented

Reply:
Do you need me to fix it anyway?

Reply:
ah I see, go ahead and leave it.

for one of these macros I left out the default case so it could support mutliple instructions, but I don't want to keep removing that